File: SEP3_UserSystem/src/main/java/com/example/SEP3_UserSystem/model/domain/unit/user/UserState.java
package com.example.SEP3_UserSystem.model.domain.unit.user;

public class UserState {
    private boolean Online;
    private String State;

    public UserState()
    {
        Online = false;
    }

    private UserState(boolean online,String state)
    {
        Online = online;
        State = state;
    }

    public void login()
    {
        Online = true;
        State = "online";
    }

    public void logoff()
    {
        Online = false;
    }

    public void setState(String state) {
        State = state;
    }

    public String getState() {
        if (Online)
        {
            return "[" + State + "]";
        }
        else
        {
            return "[offline]";
        }
    }

    public boolean isOnline() {
        return Online;
    }

    public UserState copy()
    {
        return new UserState(Online,State);
    }
}

File: SEP3_Client/Model/Unit/User/UserState.cs
namespace SEP3_Client.Model.Unit.User
{
    public class UserState
    {
        public bool Online { get; set; }
        public string State { get; set; }
        
        public UserState()
        {
            Online = false;
        }

        public void Login()
        {
            Online = true;
            State = "online";
        }

        public void Logoff()
        {
            Online = false;
        }

        public void SetState(string state) {
            State = state;
        }

        public string GetState() {
            if (Online)
            {
                return "[" + State + "]";
            }
            else
            {
                return "[offline]";
            }
        }

        public bool IsOnline() {
            return Online;
        }

        public UserState Copy()
        {
            return new UserState
            {
                Online = Online,
                State = State
            };
        }
    }
}